common/static/common/js/discussion/views/discussion_thread_list_view.js

```diff
@@ -493,28 +509,114 @@
                 return crumbs;
             };
 
-            DiscussionThreadListView.prototype.filterTopics = function(event) {
-                var items, query,
+            DiscussionThreadListView.prototype.selectOption = function(element) {
+                var activeDescendantId, activeDescendantText;
+                if (this.selectedTopic.length > 0) {
+                    this.selectedTopic.removeClass('is-focused');
+                }
+                if (element) {
+                    element.addClass('is-focused');
+                    activeDescendantId = element.parent().attr('id');
+                    activeDescendantText = element.text();
+                    this.selectedTopic = element;
+                    this.selectedTopicId = activeDescendantId;
+                    $('.forum-nav-browse-filter-input')
+                        .attr('aria-activedescendant', activeDescendantId)
+                        .val(activeDescendantText);
+                }
+            };
+
+            DiscussionThreadListView.prototype.filterInputReset = function() {
+                this.filterEnabled = true;
+                this.selectedTopicIndex = -1;
+                this.selectedTopicId = null;
+            };
+
+            DiscussionThreadListView.prototype.keyboardBinding = function(event) {
+                var $inputText = $('.forum-nav-browse-filter-input'),
+                    $filteredMenuItems = $('.forum-nav-browse-menu-item:visible'),
+                    filteredMenuItemsLen = $filteredMenuItems.length,
+                    $curOption = $filteredMenuItems.eq(0).find('.forum-nav-browse-title').eq(0),
+                    $activeOption, $prev, $next;
+
+                switch (event.keyCode) {
+                case 13:  // enter key
+                    $activeOption = $filteredMenuItems.find('.forum-nav-browse-title.is-focused');
+                    if ($inputText.val() !== '' && $inputText.focus()) {
```

Comment:
Still have `&& $inputText.focus()` here - as I understand it, this can never return false